### ISSUE TITLE
Failed to load plugin, Cannot find module "eslint/use-at-your-own-risk"

### ISSUE THREAD

**Issue by good-idea** [mantoni/eslint_d.js#183]
I'm having a problem with eslint_d being unable to load a plugin. In my case it's `@typescript/eslint-plugin` but github issue search shows that this happens in some scenarios with other plugins. Here is the most relevant issue I could find: https://github.com/typescript-eslint/typescript-eslint/issues/4210

Here is the full error & stack:

```
Error: Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js': Cannot find module 'eslint/use-at-your-own-risk'
Require stack:
- /Users/joseph/Sites/myProject/node_modules/@typescript-eslint/eslint-plugin/dist/util/getESLintCoreRule.js
- /Users/joseph/Sites/myProject/node_modules/@typescript-eslint/eslint-plugin/dist/rules/brace-style.js
- /Users/joseph/Sites/myProject/node_modules/@typescript-eslint/eslint-plugin/dist/rules/index.js
- /Users/joseph/Sites/myProject/node_modules/@typescript-eslint/eslint-plugin/dist/index.js
- /Users/joseph/Sites/myProject/node_modules/@eslint/eslintrc/dist/eslintrc.cjs
```

Running `eslint_d some-file.ts` causes this error, while `yarn eslint some-file.ts` works fine. (The latter command uses the locally installed eslint).

Here are my locally installed eslint packages:

```
    "@typescript-eslint/eslint-plugin": "^5.9.1",
    "@typescript-eslint/parser": "^5.9.1",
    "eslint-config-next": "^12.0.7",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react": "^7.28.0",
    "eslint-plugin-react-native": "^4.0.0",
```

and `eslint_d --version` says: `v7.32.0 (eslint_d v11.1.1)`

I don't really understand what's going on here.. is there any other info I can provide to help debug?

**Comment by mantoni** [comment]
Did you try `eslint_d restart` and then calling `eslint_d some-file.ts` again? Config changes are not always automatically detected in edge-case scenarios like this, and you might need to restart the daemon.

**Comment by lucasvianav** [comment]
> Did you try `eslint_d restart` and then calling `eslint_d some-file.ts` again? Config changes are not always automatically detected in edge-case scenarios like this, and you might need to restart the daemon.

I was having the same problem and this worked for me (:

**Comment by AndreiCalazans** [comment]
It seems like the issue is related to using a different eslint version than your config. My eslint_d is also using version v7 but my project is using v8. I tried using `--eslint-path` to set to the project's eslint but it hasn't worked yet. I'll post here once I find a solution.

**Comment by dastein1** [comment]
@AndreiCalazans I think you are spot on. I have the same issue with eslint_d at v7 and the project stuck at v6.

**Comment by ZLevine** [comment]
> > Did you try `eslint_d restart` and then calling `eslint_d some-file.ts` again? Config changes are not always automatically detected in edge-case scenarios like this, and you might need to restart the daemon.
> 
> I was having the exact same problem and this worked for me (:

Same here—just needed a simple `eslint_d restart` actually. I hadn't rebooted my computer since installing `eslint_d` originally and it was never working until I restarted. I'm guessing a computer reboot would've done the same (in my case at least).

**Comment by IPWright83** [comment]
@AndreiCalazans what do you mean by a different version than your config? I'm getting this too, but I'm in full control of my project setup so can change stuff around. Is there a version you can specify somewhere in the `.eslintrc.*` config? I can't see one in the docs

**Comment by AndreiCalazans** [comment]
> @AndreiCalazans what do you mean by a different version than your config? I'm getting this too, but I'm in full control of my project setup so can change stuff around. Is there a version you can specify somewhere in the `.eslintrc.*` config? I can't see one in the docs

Your project's Eslint version is set in your package.json.

I ended up using [the VSCode Eslint Language Server](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#eslint) which isnt as fast as eslint_d is but it works on my multi-project setup.

**Comment by mantoni** [comment]
In #198 we discovered a flaw with clearing the internal cache of `eslint_d`. It never really worked.

With 📦 `v12.2.0`, this feature has been re-implemented to actually do the right thing (see #200).

I hope this is finally making these problems disappear for everyone. In case new issues come up, please feel free to open a new ticket.

@AndreiCalazans It should work on multi-project setup again 😉
